Framework/CurveFitting/test/Functions/Bk2BkExpConvPVTest.h

```diff
@@ -44,5 +45,36 @@ class Bk2BkExpConvPVTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(y[0], 0.0000, 1e-4);
     TS_ASSERT_DELTA(y[50], 2.7983, 1e-4);
     TS_ASSERT_DELTA(y[99], 0.0000, 1e-4);
+
+    // When gamma is 0, `Bk2BkExpConvPV` should give same result
+    // as `BackToBackExponential`.
+    BackToBackExponential peak_b2bExp;
+    peak_b2bExp.initialize();
+    peak_b2bExp.setParameter("I", 100.0);
+    peak_b2bExp.setParameter("X0", 400.0);
+    peak_b2bExp.setParameter("A", 1.0);
+    peak_b2bExp.setParameter("B", 1.5);
+    // Attention here to the sigma parameter for `BackToBackExponential`
+    // and sigma^2 parameter for `Bk2BkExpConvPV`. We may need to change
+    // the input parameter for `Bk2BkExpConvPV` to be sigma as well.
+    peak_b2bExp.setParameter("S", sqrt(200.0));
+
+    Mantid::API::FunctionValues y_b2bExp(x);
+
+    TS_ASSERT_THROWS_NOTHING(peak_b2bExp.function(x, y_b2bExp));
+    TS_ASSERT_DELTA(y[0], y_b2bExp[0], 1e-4);
+    TS_ASSERT_DELTA(y[50], y_b2bExp[50], 1e-4);
+    TS_ASSERT_DELTA(y[99], y_b2bExp[99], 1e-4);
+
+    // We need a test for non-zero gamma since otherwise the testing
+    // for the exponential integral part is ignored. Given gamma = 1.0,
+    // We still have a peak ~400 but the values are slightly different as
+    // compared to when gamma = 0.
+    peak.setParameter("Gamma", 1.0);
```

Comment:
Great work sorting out the exponential integral, and thanks for adding the tests, but would it be possible to split this test into 3 separate tests? 
1. The original
2. Comparison with `BackToBackExponential` when` Gamma=0`
3. `Gamma > 0`

Reply:
@RichardWaiteSTFC Do you mean creating another two separate testing files like `Bk2BkExpConvPVTest1.h` and `Bk2BkExpConvPVTest2.h` to include the other two tests? Sorry I am not seeing why we need to do this, since we are just testing the single profile function?

Reply:
Sorry  - I meant to include three separate tests in the same file - so if e.g. the gamma > 0 test fails we know automatically which part of the function is broken. Let me know if you need a hand with this.

Reply:
For an example see 
https://github.com/mantidproject/mantid/blob/3a31ba7409a9cf020de27f56930ab113e2473654/Framework/CurveFitting/test/Functions/BackToBackExponentialTest.h

Reply:
So instead of having one function inside the test suite
`void test_functionCalculator()`
It'd be good to have at least two with informative names
`void test_function_evaluation_gamma_equal_to_zero()`
`void test_function_evaluation_gamma_greater_than_zero()`